packages/react/src/hooks/useFocusWithin.ts

```diff
@@ -0,0 +1,22 @@
+import type { RefObject } from 'react';
+import { useState } from 'react';
+
+import { useEventListener } from './useEventListener';
+
+/**
+ * Returns true if the given ref is currently focused or contains the currently focused element.
+ * @param ref Reference to the element to check.
+ * @returns True if the given ref is currently focused or contains the currently focused element.
+ */
+export function useFocusWithin<T extends HTMLElement>(
+  ref: RefObject<T>,
```

Comment:
I did try that, but I couldn't make the test pass for some reason. I can take a deeper look into it though.

Reply:
Ah, gimme a headsup if you need a second pair of eyes 👀

Reply:
The problem was that an update of `ref.current` didn't trigger a rerender since it is a mutable property, so I had to add an additional `useEffect` in the test.